program-runtime/src/loaded_programs.rs

```diff
@@ -328,6 +328,10 @@ impl LoadedPrograms {
                         Ordering::Relaxed,
                     );
                     second_level.swap_remove(entry_index);
+                } else if existing.is_tombstone() && !entry.is_tombstone() {
+                    // The old entry is tombstone and the new one is not. Let's give the new entry
+                    // a chance.
+                    second_level.swap_remove(entry_index);
```

Comment:
for the record: this newly-added instance of `.swap_remove()` is fixed at #32973